packages/material-ui-lab/src/TreeView/TreeView.js

```diff
@@ -483,6 +483,11 @@ const TreeView = React.forwardRef(function TreeView(props, ref) {
     }
   }, [expanded, childrenCalculated, isExpanded]);
 
+  React.useEffect(() => {
```

Comment:
Does this work during render as well? Basically use derived state pattern (remove the React.useEffect) instead of a side-effect pattern?

Reply:
Yes that works.  I did not know that you could setState in render.  Pushing commit now